backend/httpclient/config/max_bytes_reader.go

```diff
@@ -0,0 +1,66 @@
+package config
+
+import (
+	"errors"
+	"fmt"
+	"io"
+)
+
+// Similar implementation to http/net MaxBytesReader
+// https://pkg.go.dev/net/http#MaxBytesReader
+// What's happening differently here, is that the field that
+// is limited is the response and not the request, thus
+// the error handling/message needed to be accurate.
+
+// ErrResponseBodyTooLarge indicates response body is too large
+var ErrResponseBodyTooLarge = errors.New("http: response body too large")
+
+// MaxBytesReader is similar to io.LimitReader but is intended for
+// limiting the size of incoming request bodies. In contrast to
+// io.LimitReader, MaxBytesReader's result is a ReadCloser, returns a
+// non-EOF error for a Read beyond the limit, and closes the
+// underlying reader when its Close method is called.
+//
+// MaxBytesReader prevents clients from accidentally or maliciously
+// sending a large request and wasting server resources.
+func MaxBytesReader(r io.ReadCloser, n int64) io.ReadCloser {
+	return &maxBytesReader{r: r, n: n}
+}
+
+type maxBytesReader struct {
+	r   io.ReadCloser // underlying reader
```

Comment:
Can this just be `io.Reader` instead since it doesn't need to close anything? I realize this is copied from Grafana, but wonder if it would simplify a bit 🤔 No biggie

Reply:
But res.Body is an `io.ReadCloser` which itself its just `io.Reader` - so technically it should be okay or am I missing something else?

Reply:
yeah, [here](https://github.com/grafana/grafana-plugin-sdk-go/pull/917/files#diff-a2616ca6ad766c7df86e8ae6a0a8a9c67af732f4dff079ba354a19c32992890eR27) we are doing `res.Body = MaxBytesReader(res.Body, limit)` so we need to implement the `Close` function